packages/compass-preferences-model/index.d.ts

```diff
@@ -1,15 +1,31 @@
-declare class CompassPreferencesModel {
-  fetch: () => void;
-  save: () => void;
-  set: (
-    key: string,
-    value: unknown
-  ) => void;
-};
-export default CompassPreferencesModel;
+import { preferencesIpc } from '.';
 
 export enum THEMES {
   DARK = 'DARK',
   LIGHT = 'LIGHT',
   OS_THEME = 'OS_THEME'
-};
+}
+
+export type UserPreferences = {
+  showedNetworkOptIn: boolean; // Has the settings dialog has been shown before.
+  autoUpdates: boolean;
+  enableMaps: boolean;
+  trackErrors: boolean;
+  trackUsageStatistics: boolean;
+  enableFeedbackPanel: boolean;
+  theme: THEMES;
+}
+
+export type GlobalPreferences = UserPreferences; // TODO: extend with global preferences.
+
+declare class Preferences {
+  constructor(basepath?: string): void;
+  fetchPreferences(): Promise<GlobalPreferences>;
+  savePreferences(attributes: GlobalPreferences): Promise<GlobalPreferences>;
+  getPreferences(): Promise<GlobalPreferences>;
+  getConfigurableUserPreferences(): Promise<UserPreferences>;
+}
+
+export { preferencesIpc };
```

Comment:
This is untyped … do we maybe want to convert the implementation files to TS after this PR? That would also make it easier to ensure that `preferencesProps` contains the right things.

Reply:
Yes!